<?xml version="1.0"?>
<launch>
  <!-- Vehicle Control Node Launch File -->
  
  <!-- Arguments -->
  <arg name="config_file" default="$(find vehicle_control)/config/control.yaml" />
  <arg name="output" default="screen" />
  
  <!-- Vehicle Control Node -->
  <node name="vehicle_control_node" 
        pkg="vehicle_control" 
        type="control.py" 
        output="$(arg output)"
        respawn="false">
    
    <!-- Load parameters from YAML file -->
    <rosparam command="load" file="$(arg config_file)" />
    
    <!-- Topic remapping (if needed) -->
    <!-- <remap from="/vehicle_state" to="/carla/vehicle_state" /> -->
    <!-- <remap from="/reference_state" to="/traj_server/reference_state" /> -->
    <!-- <remap from="/control_cmd" to="/carla/control_cmd" /> -->
    
  </node>
  
</launch>
